hs_core/hs_rdf.py

```diff
@@ -0,0 +1,358 @@
+                term_values.append((field_term, field_value))
+        return term_values
+
+    @classmethod
+    def get_class_term(cls):
+        """
+        return a term mapped by the @rdf_terms decorator or if the deocorator is not provided, create an
+        HSTERMS.{cls.__name__}
+        """
+        return cls.rdf_term if hasattr(cls, 'rdf_term') else getattr(HSTERMS, cls.__name__)
+
+    @classmethod
+    def get_field_term(cls, field_name):
+        """
+        Given a class field_name, return a term mapped by the @rdf_terms decoroator or if the decorator is
+        not provided, create an HSTERMS.{field_name}
+        """
+        field_term_attr = field_name + '_rdf_term'
+        if hasattr(cls, field_term_attr):
+            return getattr(cls, field_term_attr)
+        else:
+            return getattr(HSTERMS, field_name)
+
+    @classmethod
+    def ingest_rdf(cls, graph, subject, content_object):
+        """Default implementation that ingests by convention"""
+        term = cls.get_class_term()
+        if not term:
+            metadata_nodes = [subject]
+        else:
+            metadata_nodes = graph.objects(subject=subject, predicate=term)
+        for metadata_node in metadata_nodes:
+            value_dict = {}
+            for field in cls._meta.fields:
+                if cls.ignored_fields and field.name in cls.ignored_fields:
+                    continue
+                field_term = cls.get_field_term(field.name)
+                val = graph.value(metadata_node, field_term)
+                if val is not None:
+                    value_dict[field.name] = str(val)
+            if value_dict:
+                cls.create(content_object=content_object, **value_dict)
+
+
+def rdf_terms(class_term, **field_terms):
+    """
+    Deocorator for mapping a class and fields to RDF terms.  The first parameter maps the class to your specified
+    term.  field_terms are key/values with the key matching a class's field and the value being the mapped term.
+
+    :raises ValidationError when a field_term key does not match a field on the class"""
+    def decorator(obj):
+        obj.rdf_term = class_term
+        for k, v in field_terms.items():
+            if not hasattr(obj, k):
+                raise ValidationError("field {} not found".format(k))
+            setattr(obj, k + '_rdf_term', v)
+        return obj
+    return decorator
+
+
+class HydroPrettyXMLSerializer(Serializer):
+    """Same as PrettyXMLSerializer but with stripped out node ids"""
+
+    def __init__(self, store, max_depth=3):
+        super(HydroPrettyXMLSerializer, self).__init__(store)
+        self.forceRDFAbout = set()
+
+    def serialize(self, stream, base=None, encoding=None, **args):
+        self.__serialized = {}
+        store = self.store
+        self.base = base
+        self.max_depth = args.get("max_depth", 3)
+        assert self.max_depth > 0, "max_depth must be greater than 0"
```

Comment:
I am wondering whether assert should be replaced by a formal check since assert will be ignored in non-debug mode.

Reply:
This serializer is a near copy for the PrettyXMLSerializer, it behaves the same except the node ids have been stripped out.  I'd prefer to not touch this serializer anymore than we have to.

Reply:
Looks like the rest of your comments in this file are on this serializer.  I'd prefer not to touch it because it is mostly just a copy of another serializer, but I'm open to talking about it.

Reply:
Hmm. I am not sure that assert is ignored in non-debug mode. I had to avoid checking it in the access control system via guard clauses for this reason.